lib/quiet_quality/executors/concurrent_executor.rb

```diff
@@ -0,0 +1,20 @@
+module QuietQuality
+  module Executors
+    class ConcurrentExecutor < BaseExecutor
+      def execute!
+        threads = pipelines.map { |pipeline| threaded_pipeline(pipeline) }
+        threads.each(&:join)
```

Comment:
I forgot how simple this _particular_ thing was to do with ruby - process-parallel code is really easy! The awkward part will be when we want the pipelines to start writing their pass/fail status as the results land, which will involve having the threads write to a Queue, and using an extra thread to watch the queue and write to the stderr and pipelines complete.